OpenRA.Game/GameRules/Ruleset.cs

```diff
@@ -87,16 +87,26 @@ public class Ruleset
 
 		public IEnumerable<KeyValuePair<string, MusicInfo>> InstalledMusic { get { return Music.Where(m => m.Value.Exists); } }
 
-		static IReadOnlyDictionary<string, T> MergeOrDefault<T>(string name, IReadOnlyFileSystem fileSystem, IEnumerable<string> files, MiniYaml additional,
-			IReadOnlyDictionary<string, T> defaults, Func<MiniYamlNode, T> makeObject)
+		static IReadOnlyDictionary<string, T> MergeOrDefault<T>(string name,
+			IReadOnlyFileSystem fileSystem,
+			IEnumerable<string> files,
+			MiniYaml additional,
+			IReadOnlyDictionary<string, T> defaults,
+			Func<MiniYamlNode, T> makeObject,
+			Func<MiniYamlNode, bool> filterNode = null)
 		{
 			if (additional == null && defaults != null)
 				return defaults;
 
-			var result = MiniYaml.Load(fileSystem, files, additional)
-				.ToDictionaryWithConflictLog(k => k.Key.ToLowerInvariant(), makeObject, "LoadFromManifest<" + name + ">");
+			var yamlNodes = MiniYaml.Load(fileSystem, files, additional) as IEnumerable<MiniYamlNode>;
 
-			return new ReadOnlyDictionary<string, T>(result);
+			// Optionally, the caller can filter out elements from the loaded set of nodes. Default behavior is unfiltered.
+			if (filterNode != null)
+			{
```

Comment:
No braces around single-line statements, please. ^^

Reply:
Thanks - old habit. I'll see if I can find a linter or editor extension to help me catch these.